CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+# Contributing to fccodegen
+
+## Installation
+
+Users install `fccodegen` using pip.
+For instructions on installing as a user, see [README.md](README.md)
+
+Developers should
+
+1. Create a virtual environment, using e.g. virtualenv or conda. Make sure to use Python3 (e.g. `$ virtualenv venv -p python3`).
+2. Activate the environment (e.g. `$ source venv/bin/activate`)
+3. Install the developer requirements into the virtual environment: `pip install -r dev-requirements/dev.txt`
+4. Run the tests: `$ python -m pytest`.
+
+
+
+### Requirements
+
+There are two lists of requirements.
+
+1. User requirements, specified in `setup.py`.
+2. Developer requirements, specified in the `dev-requirements` directory.
+
+User requirements specify minimum versions of each dependency, and can be used in an existing Python ecosystem.
+Developer requirement specify "pinned" versions of each dependency, and should be installed inside a virtual environment.
+Having pinned versions ensures different developers get consistent results.
+Continuous integration testing happens with user requirements.
+
+Using a virtualenv for development also helps you notice when requirements are missing from the developer requirements.
+Similarly, on CI testing anything not listed in the user requirements will cause the tests to fail.
+
+**User requirements** are listed in `setup.py`.
+They are divided into base dependencies (listed in `install_requires`) and `test` dependencies (listed in `extras_require[test]`).
+Users install these requirements automatically when they `pip install fccodegen`.
+
+**Developer requirements** are listed in `base.in`, `test.in`, and `dev.in` (where `test` requires `base`, while `dev` requires `base` and `test`).
+These `.in` files are compiled into `.txt` files for `pip` using [isort](https://pypi.org/project/isort/) and [pip-tools](https://pypi.org/project/pip-tools/).
```

Comment:
Actually only pip-tools is used for this. isort is another style checker that checks your import lines.